src/models/wasteWater/story/WasteWaterLegacySites.tsx

```diff
@@ -0,0 +1,50 @@
+import { legacyLocations } from './constants';
+import { filterByDateRange, getMaxDateRange, useWasteWaterData } from './WasteWaterSamplingSitesHooks';
+import React, { useState } from 'react';
+import { DateRangeSelector, SpecialDateRangeSelector } from '../../../data/DateRangeSelector';
+import Loader from '../../../components/Loader';
+import DateRangePicker from '../../../components/DateRangePicker';
+import { GridCell, PackedGrid } from '../../../components/PackedGrid';
+import { WasteWaterLocationTimeWidget } from '../WasteWaterLocationTimeWidget';
+import { ShowMoreButton } from '../../../helpers/ui';
+
+export const WasteWaterLegacySites = () => {
+  const wasteWaterData = useWasteWaterData();
+  const [dateRangeSelector, setDateRangeSelector] = useState<DateRangeSelector>(
+    new SpecialDateRangeSelector('Past6M')
+  );
+
+  if (!wasteWaterData) {
+    return <Loader />;
+  }
+
+  const dataInTimeRange = filterByDateRange(wasteWaterData, dateRangeSelector.getDateRange()).filter((ds) => legacyLocations.has(ds.location));
```

Comment:
Instead of duplicating the function, this should be a wrapper into a common function and the filter should be passed as a lamba.